protocol/storage/miner.go

```diff
@@ -476,6 +487,32 @@ func (sm *Miner) processStorageDeal(proposalCid cid.Cid) {
 	}
 }
 
+func (sm *Miner) validatePieceCommitments(ctx context.Context, deal *storagedeal.Deal, rootIpldNode format.Node, serv format.NodeGetter) error {
+	pieceReader, err := uio.NewDagReader(ctx, rootIpldNode, serv)
+	if err != nil {
+		return err
+	}
+
+	// Generating the piece commitment is a computationally expensive operation and can take
+	// many minutes depending on the size of the piece.
+	pieceCommitmentResponse, err := proofs.GeneratePieceCommitment(proofs.GeneratePieceCommitmentRequest{
+		PieceReader: pieceReader,
+		PieceSize:   types.NewBytesAmount(deal.Proposal.Size.Uint64()),
+	})
```

Comment:
This will have a very noticeable operational effect for miners - those doing automated large deals will see a big CPU spike. We must make sure they are not surprised by this.